modin/experimental/pandas/__init__.py

```diff
@@ -45,11 +45,8 @@
     read_custom_text,
     read_pickle_distributed,
     read_sql,
-    to_pickle_distributed,
 )
 
-setattr(DataFrame, "to_pickle_distributed", to_pickle_distributed)  # noqa: F405
-
```

Comment:
why? it seems like it breaks the old code:
```python
from modin.config import IsExperimental
IsExperimental.put(True)

import modin.experimental.pandas as pd
pd.DataFrame(...).to_pickle_distributed() # fails
```

Reply:
I would expect the new behavior via `._exp` to be adding a new way of invoking experimental method, but not replacing the previous one

Reply:
Yes, it breaks. Initially, this was done without taking into account the case when the experimental mode should work in conjunction with the regular one. Updating attributes during import makes context switching very difficult.

Since this is experimental code, I thought that such a breaking change would not be a big problem.

Reply:
> Updating attributes during import makes context switching very difficult

why? shouldn't `enable_exp_mode` context do nothing if the experimental mode is already enabled?

Reply:
> enable_exp_mode context do nothing if the experimental mode is already enabled?

Yes, but what if `import modin.experimental.pandas` is made in this context? The expected behavior would be that when exiting the context, no side effects remain.

Reply:
Well, it still should be fine then. If I understand correctly, after we leave the context, the experimental methods of a DataFrame will still be accessible but not functioning, which seems like a pretty logical behavior
```python
with enable_exp_mode():
    import modin.experimental.pandas as pd
    pd.DataFrame([]).to_pickle_distributed() # works

pd.DataFrame([]).to_pickle_distributed() # raises: not in experimental mode
```

Reply:
I brought back this function, but added a warning about future removal and an alternative to use. I hope this suits you.